milli/src/update/words_level_positions.rs

```diff
@@ -102,13 +101,23 @@ impl<'t, 'u, 'i> WordsLevelPositions<'t, 'u, 'i> {
         // in the prefix FST previously constructed.
         let prefix_fst = self.index.words_prefixes_fst(self.wtxn)?;
         let db = self.index.word_level_position_docids.remap_data_type::<ByteSlice>();
-        for result in db.iter(self.wtxn)? {
-            let ((word, level, left, right), data) = result?;
-            if level == TreeLevel::min_value() {
-                let automaton = automaton::Str::new(word).starts_with();
-                let mut matching_prefixes = prefix_fst.search(automaton).into_stream();
-                while let Some(prefix) = matching_prefixes.next() {
-                    let prefix = str::from_utf8(prefix)?;
+        // iter over all prefixes in the prefix fst.
+        let mut word_stream = prefix_fst.stream();
+        while let Some(prefix_bytes) = word_stream.next() {
+            let prefix = str::from_utf8(prefix_bytes).expect(&format!(
+                "prefix {:?}, comming from prefix FST, is not a valid UTF-8 string",
+                prefix_bytes
+            ));
```

Comment:
Doing this will force a string creation at every single loop, don't do that.
Please `map_err` with a format and the related error and use the `?` operator.

Reply:
Use the `?` instead of unwraps as it is better to use "pure" functions instead of functions that can panic everywhere.

Reply:
@Kerollmops @MarinPostma I can suggest:
```suggestion
            let prefix = str::from_utf8(prefix_bytes).map_err(|e| {
                SerializationError::Decoding { db_name: Some(WORDS_PREFIXES_FST_KEY) }
            })?;
```
We return an error (Internal) and we know which database is "corrupted".
What do you think?

Reply:
So the error is  reported to the user, but the program continues to run crippled? @gmourier what do you think?

Reply:
Hello guys! Not sure to understand everything here. 

We now check that the prefix is encoded in UTF-8, if it's not the case we raise an error but MeiliSearch doesn't crash and keeps working, right? 

I have some questions to better understand 🤓.

- What does it imply for the user? 
- What can he do about that?
- Do we have this type of behavior anywhere else? (Having a corrupted DB but continuing to operate for a search request, we talked about that some times ago, is it related?)
- Why can this only happen on large datasets?